### PR TITLE
614204: Add message distributor container module

### PR DESCRIPTION
**Pull request by rorytorneymf**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=614204

### CODE REVIEW

File: worker-message-prioritization-distribution-container/pom.xml [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+                                    <shutdown>500</shutdown>
+                                </wait>
+                                <log>
+                                    <enabled>true</enabled>
+                                </log>
+                            </run>
+                        </image>
+
+                        <image>
+                            <name>${dockerWorkerFrameworkOrg}worker-message-prioritization-distribution${dockerProjectVersion}</name>
```
Comment (rorytorneymf) [inline worker-message-prioritization-distribution-container/pom.xml:168]: I don't see any workerframework org currently on dockerhub so we might need to create one:

https://hub.docker.com/search?q=workerframework

File: worker-message-prioritization-distribution-container/pom.xml [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+                                    <Git.Repo>https://github.com/WorkerFramework/worker-message-prioritization</Git.Repo>
+                                    <Git.Branch>${git.branch}</Git.Branch>
+                                    <Git.Commit>${git.revision}</Git.Commit>
+                                </labels>
+                                <healthCheck>
+                                    <!--
+                                        Check if the Java process is running. This also indirectly verifies that the connection to
+                                        RabbitMQ is ok, since the Java process will exit if it cannot connect to RabbitMQ.
+                                    -->
+                                    <cmd>/maven/healthcheck/healthcheck.sh || exit 1</cmd>
```
Comment (rorytorneymf) [inline worker-message-prioritization-distribution-container/pom.xml:184]: Although healthcheck.sh contains only a small command, it needs to be in a script to work. When I tried to put it inline in the pom.xml here:

```
<cmd>pgrep -f "java.*worker-message-prioritization-distribution.*jar" > /dev/null || exit 1</cmd>
```

the healthcheck always reported healthy, even when it should not have.

File: worker-message-prioritization-distribution-container/src/test/java/com/github/workerframework/workermessageprioritization/redistribution/LowLevelDistributorIT.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import com.rabbitmq.client.Connection;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.concurrent.TimeoutException;
+
+public final class LowLevelDistributorIT extends DistributorTestBase {
```
Comment (rorytorneymf) [inline worker-message-prioritization-distribution-container/src/test/java/com/github/workerframework/workermessageprioritization/redistribution/LowLevelDistributorIT.java:30]: This test is almost the same as [LowLevelDistributorIT.java](https://github.com/WorkerFramework/worker-message-prioritization/blob/main/worker-message-prioritization-distribution/src/test/java/com/github/workerframework/workermessageprioritization/redistribution/LowLevelDistributorIT.java), except that it is not calling any domain objects, its just sending messages to rabbit. I don't know if we still want to keep both sets of tests.

File: worker-message-prioritization-distribution-container/pom.xml (hunk 1 above)
Comment (andyreidz) [reply worker-message-prioritization-distribution-container/pom.xml:168]: https://hub.docker.com/orgs/workerframework

File: worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/config/MessageDistributorConfig.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        return rabbitMQMgmtPassword;
+    }
+
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this)
+            .add(WORKER_MESSAGE_PRIORITIZATION_RABBITMQ_HOST, rabbitMQHost)
+            .add(WORKER_MESSAGE_PRIORITIZATION_RABBITMQ_PORT, rabbitMQPort)
+            .add(WORKER_MESSAGE_PRIORITIZATION_RABBITMQ_USERNAME, rabbitMQUsername)
+            .add(WORKER_MESSAGE_PRIORITIZATION_RABBITMQ_PASSWORD, "<HIDDEN>")
```
Comment (andyreidz) [inline worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/config/MessageDistributorConfig.java:133]: 👍

**Review by andyreidz — CHANGES_REQUESTED** [review]
Looking good, some comments on the ENV names and where to put the doc.

File: README.md [hunk 5]
```diff
@@ -29,3 +29,42 @@ messages from a staging queue and publish them to the worker input queue. The se
 shovel plugin to achieve the same goal. Ultimately a much simpler approach but with the requirement of an additional 
 RabbitMq plugin. **During the POC phase some instability was observed with this plugin which should be revisted.**
 
+## Configuration
+
+### Environment Variables
+
+* `WORKER_MESSAGE_PRIORITIZATION_RABBITMQ_HOST`
```
[the remark this answers is not in the document]
Comment (rorytorneymf) [reply README.md]: I've gone with the CAF_ prefix for now

File: worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/MessageDistributor.java [hunk 6]
```diff
@@ -46,6 +53,8 @@ protected Set<DistributorWorkItem> getDistributorWorkItems() {
                             q.getMessages() > 0 &&
                                     q.getName().startsWith(targetQueue.getName() + LOAD_BALANCED_INDICATOR))
                     .collect(Collectors.toSet());
+
+            LOGGER.info("Filtered the list of queues from the RabbitMQ API to contain only the staging queues: {}", stagingQueues);
```
[the remark this answers is not in the document]
Comment (rorytorneymf) [reply worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/MessageDistributor.java]: I've changed these to debug now.

**Review by andyreidz — APPROVED** [review]
All good, thanks!

**Pull request merged**